src/main/java/com/bobocode/hoverla/bring/context/BeanInitializer.java

```diff
@@ -1,13 +1,40 @@
 package com.bobocode.hoverla.bring.context;
 
-import java.util.Map;
+import com.google.common.collect.Table;
+import lombok.extern.slf4j.Slf4j;
 
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * Class, responsible for triggering bean initialization.
+ * <br>
+ * Its main aim is to pass all dependencies of a particular {@link BeanDefinition}
+ * to its {@link BeanDefinition#instance(BeanDefinition...)} method.
+ */
+@Slf4j
 public class BeanInitializer {
 
+
     /**
-     * Instantiates beans and injects dependencies in them
+     * Triggers instantiation of all beans by passing their required dependencies.
+     * @param beanDefinitionsTable {@link Table} with all {@link BeanDefinition} objects handled by current context.
      */
-    void initialize(Map<String, BeanDefinition> beanDefinitions) {
+    public void initialize(Table<String, Class<?>, BeanDefinition> beanDefinitionsTable) {
+        log.debug("Bean initialization started");
+        Collection<BeanDefinition> beanDefinitions = beanDefinitionsTable.values();
+
+        for (BeanDefinition beanDefinition : beanDefinitions) {
+            log.trace("Resolving dependencies for bean definition: {} - {}", beanDefinition.name(), beanDefinition.type());
+            BeanDefinition[] beanDependencies = beanDefinition.dependencies()
+                    .entrySet()
+                    .stream()
+                    .map(keyPair -> beanDefinitionsTable.get(keyPair.getKey(), keyPair.getValue()))
+                    .filter(Objects::nonNull)
```

Comment:
(nit) seems like we don't need this check, because dependency missing will be handled in validator